src/ray/util/internal/stream_redirection_utils.h

```diff
@@ -0,0 +1,45 @@
+// Copyright 2025 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <memory>
+
+#include "ray/util/pipe_logger.h"
+#include "ray/util/scoped_dup2_wrapper.h"
+
+namespace ray::internal {
+
+struct RedirectionHandleWrapper {
+  RedirectionFileHandle redirection_file_handle;
+  // Used for restoration.
+  std::unique_ptr<ScopedDup2Wrapper> scoped_dup2_wrapper;
+};
+
+// Util functions to redirect stdout / stderr stream based on the given redirection option
+// [opt]. This function is _NOT_ thread-safe.
+RedirectionHandleWrapper RedirectStreamImpl(MEMFD_TYPE_NON_UNIQUE stream_fd,
+                                            const StreamRedirectionOption &opt);
+
+// Synchronize on redirection handler in blocking style.
```

Comment:
the comment as-written is not very helpful to an uninformed reader because it's basically describing the implementation detail. prefer to describe the functionality and important semantic.

For example something like this (I'm partially making it up): "Synchronizes any buffered output to the redirection handle wrapper. After this function returns, <insert important information. This should be called <insert important information>."

Reply:
Reword a little to include (1) flush; (2) restore stream.